src/surge-fx/SurgeFXProcessor.cpp

```diff
@@ -492,6 +572,11 @@ void SurgefxAudioProcessor::setStateInformation(const void *data, int sizeInByte
             effectNum = xmlState->getIntAttribute("fxt", fxt_delay);
             resetFxType(effectNum, false);
 
+            oscPortIn = xmlState->getIntAttribute("oscpin", 0);
+            oscStartIn = xmlState->getBoolAttribute("oscin", false);
+            // start OSC, if variables merit it
+            oscCheckStartup = true;
```

Comment:
Shouldn't this be `oscCheckStartup = oscPortIn > 0 && oscStartIn`?

Reply:
Those are checked further downstream (in tryLazyOscStartupFromStreamedState(). This code is copied from Surge XT, and was written mostly by Evil, and I haven't modified it much at all.